controllers/demo/opnidemo_controller.go

```diff
@@ -161,18 +164,33 @@ func (r *OpniDemoReconciler) reconcileInfraStack(
 	req ctrl.Request,
 	opniDemo *v1alpha1.OpniDemo,
 ) (ctrl.Result, error) {
-	objects := demo.MakeInfraStackObjects(opniDemo)
-	objects = append(objects, demo.BuildRancherLoggingCrdHelmChart())
+	objects := []client.Object{}
+
+	if opniDemo.Spec.Components.Infra.DeployHelmController {
+		objects = append(objects, demo.BuildHelmControllerObjects(opniDemo)...)
+	}
+
+	if opniDemo.Spec.Components.Infra.DeployNvidiaPlugin {
+		objects = append(objects, demo.BuildNvidiaPlugin(opniDemo))
+	}
+
 	for _, object := range objects {
 		object.SetNamespace(opniDemo.Namespace)
 		if err := r.Get(ctx, client.ObjectKeyFromObject(object), object); errors.IsNotFound(err) {
-			r.Log.Info("creating resource", "name", client.ObjectKeyFromObject(object))
-			if err := ctrl.SetControllerReference(opniDemo, object, r.Scheme); err != nil {
+			r.log.Info("creating resource", "name", client.ObjectKeyFromObject(object))
+			if err := ctrl.SetControllerReference(opniDemo, object, r.scheme); err != nil {
 				return ctrl.Result{}, err
 			}
 			return ctrl.Result{Requeue: true}, r.Create(ctx, object)
 		} else if err != nil {
 			return ctrl.Result{}, err
+		} else {
+			if ds, ok := object.(*appsv1.DaemonSet); ok {
+				if ds.Status.NumberReady != ds.Status.DesiredNumberScheduled {
+					opniDemo.Status.Conditions = append(opniDemo.Status.Conditions,
+						fmt.Sprintf("Waiting for daemonset %s to become ready", ds.Name))
+				}
+			}
```

Comment:
We only check for the DaemonSets?

Reply:
This is just checking the nvidia daemonset and helm controller resources, I think the only workload there worth checking the status of is the nvidia daemonset, unless you have any other suggestions